app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -97,7 +97,7 @@ def queue_name_for_ems(ems)
       return "generic" if ems.kind_of?(Host) && ems.acts_as_ems?
 
       return ems unless ems.kind_of?(ExtManagementSystem)
```

Comment:
I'm not sure exactly what case this is trying to handle...why would you pass in something other than an ems to `queue_name_for_ems`?
There is a spec test for it https://github.com/ManageIQ/manageiq/blob/master/spec/models/mixins/per_ems_worker_mixin_spec.rb#L14 I just don't know why that is desired behavior